doi/src/main/java/ca/nrc/cadc/doi/DoiInitAction.java

```diff
@@ -251,12 +264,76 @@ private static MultiValuedProperties getConfig(boolean verify) {
                 sb.append("OK");
             }
         }
-        log.info(sb.toString());
 
         if (!ok) {
-            throw new IllegalStateException("incomplete config: " + sb.toString());
+            throw new IllegalStateException("incomplete config: " + sb);
         }
         return props;
     }
 
+    // check that the DOI parent node path, configured with the V0SPACE_PARENT_PATH_KEY property,
+    // exists and has the expected properties.
+    private static void checkParentFolders() {
+        MultiValuedProperties config = getConfig();
+        URI vospaceResourceID = URI.create(config.getFirstPropertyValue(DoiInitAction.VAULT_RESOURCE_ID_KEY));
+        String parentPath = config.getFirstPropertyValue(PARENT_PATH_KEY);
+
+        Subject adminSubject = SSLUtil.createSubject(new File("/config/doiadmin.pem"));
+        adminSubject = AuthenticationUtil.augmentSubject(adminSubject);
+        String adminUsername = getUsername(adminSubject);
+
+        VOSpaceClient vosClient = new VOSpaceClient(vospaceResourceID);
+        String currentPath = "";
+        String[] paths = parentPath.split("/");
+        for (String path : paths) {
+            // skip first empty path if parentPath begins with a /
+            if (!StringUtil.hasText(path)) {
+                continue;
+            }
+            currentPath = String.format("%s/%s", currentPath, path);
+
+            Node node;
+            try {
+                node = vosClient.getNode(currentPath);
+            } catch (ResourceNotFoundException e) {
+                throw new IllegalStateException(String.format("node %s not found", path));
+            } catch (Exception e) {
+                throw new IllegalStateException(String.format("node %s error because %s", path, e.getMessage()));
+            }
+
+            // confirm it's a ContainerNode
+            if (!(node instanceof ContainerNode)) {
+                throw new IllegalStateException(String.format("node %s is not a ContainerNode", path));
+            }
+            ContainerNode containerNode = (ContainerNode) node;
+
+            // check node owner
+            String ownerID = containerNode.ownerDisplay;
+            if (!adminUsername.equals(ownerID)) {
+                throw new IllegalStateException(String.format("node %s owner %s doesn't match configured admin user %s", path, ownerID, adminUsername));
+            }
+
+            // check node has public access
+            if (!containerNode.isPublic) {
+                throw new IllegalStateException(String.format("node %s must have isPublic set to true", path));
+            }
+
+            // check inheritPermissions is true (does inheritPermissions need to be true?)
+            if (!containerNode.inheritPermissions) {
+                throw new IllegalStateException(String.format("node %s must have inheritPermissions set to true", path));
+            }
+        }
+    }
+
+    private static String getUsername(Subject subject) {
```

Comment:
Hmm. This is almost like `AuthenticationUtil.getUseridsFromSubject()` but not quite...

Reply:
Yeah close, but just not quite.